pkg/kubectl/cmd/config/create_cluster.go

```diff
@@ -110,26 +110,26 @@ func (o createClusterOptions) run() error {
 }
 
 // cluster builds a Cluster object from the options
-func (o *createClusterOptions) modifyCluster(existingCluster clientcmdapi.Cluster) clientcmdapi.Cluster {
+func (o *createClusterOptions) modifyCluster(cmd *cobra.Command, existingCluster clientcmdapi.Cluster) clientcmdapi.Cluster {
 	modifiedCluster := existingCluster
 
-	if o.server.Provided() {
-		modifiedCluster.Server = o.server.Value()
+	if o.server != "" {
+		modifiedCluster.Server = o.server
 	}
-	if o.apiVersion.Provided() {
-		modifiedCluster.APIVersion = o.apiVersion.Value()
+	if o.apiVersion != "" {
+		modifiedCluster.APIVersion = o.apiVersion
 	}
-	if o.insecureSkipTLSVerify.Provided() {
-		modifiedCluster.InsecureSkipTLSVerify = o.insecureSkipTLSVerify.Value()
+	if o.insecureSkipTLSVerify {
+		modifiedCluster.InsecureSkipTLSVerify = o.insecureSkipTLSVerify
```

Comment:
@deads2k @liggitt this right here might be wrong. As explicitly setting it to false after having it true won't undo anything...   Am I getting that wrong up above too?  (Yes I can fix it)